sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ExecutingSqlQueries_Net46.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Data.SqlClient;
+using System.Data.Odbc;
+using System.Data.OracleClient;
+using System.Data.SqlServerCe;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        private const string ConstantQuery = "";
+
+        public void SqlCommands(SqlConnection connection, SqlTransaction transaction, string query)
+        {
+            SqlCommand command;
+            command = new SqlCommand(); // Compliant
+            command = new SqlCommand(""); // Compliant
+            command = new SqlCommand(ConstantQuery); // Compliant
+            command = new SqlCommand(query); // Noncompliant {{Make sure that executing SQL queries is safe here.}}
+//                    ^^^^^^^^^^^^^^^^^^^^^
+            command = new SqlCommand(query, connection); // Noncompliant
+            command = new SqlCommand(query, connection, transaction); // Noncompliant
+            command = new SqlCommand(query, connection, transaction, SqlCommandColumnEncryptionSetting.Enabled); // Noncompliant
```

Comment:
Test case: _SqlCommand_ constructor with two params and constant string as first parameter -> compliant